### PR TITLE
[sw/silicon_creator, crypto] Reject too-large signatures in RSA

### PR DESCRIPTION
**Pull request by jadephilipoom**
According to spec, we're supposed to reject signatures that are not strictly smaller than the modulus. (We already restrict the size, but it's possible that `n <= sig < 2^3072`.) This PR:
- Adds a regression test to the default test set that checks we reject large signatures
- Adds a "reverse memcmp" function to `memory.h` that we can use to arithmetically compare little-endian buffers
- Changes both the sigverify and generic cryptolib RSA implementations to reject too-large signatures

I'm not sure that it's an exploitable issue if we don't perform this check, but it does violate the spec, and our previous implementation did incorrectly accept the signature in the regression test.

### CODE REVIEW

File: sw/device/lib/base/memory.c [hunk 1]
```diff
@@ -36,13 +36,13 @@ void *memset(void *dest, int value, size_t len) {
...
 #endif  // !defined(HOST_BUILD)
 
-#if !defined(HOST_BUILD)
 enum {
   kMemCmpEq = 0,
   kMemCmpLt = -42,
   kMemCmpGt = 42,
 };
 
+#if !defined(HOST_BUILD)
```
Comment (mcy) [inline sw/device/lib/base/memory.c:45]: `ifndef`?

**Review by mcy — APPROVED** [review]

File: sw/device/silicon_creator/lib/sigverify_mod_exp_ibex.c [hunk 2]
```diff
@@ -166,6 +166,12 @@ static void calc_r_square(const sigverify_rsa_key_t *key,
 rom_error_t sigverify_mod_exp_ibex(const sigverify_rsa_key_t *key,
                                    const sigverify_rsa_buffer_t *sig,
                                    sigverify_rsa_buffer_t *result) {
+  // Reject the signature if it is too large (n <= sig): RFC 8017, section
+  // 5.2.2, step 1.
+  if (memrcmp(key->n.data, sig->data, kSigVerifyRsaNumBytes) <= 0) {
```
Comment (alphan) [inline sw/device/silicon_creator/lib/sigverify_mod_exp_ibex.c:171]: I think we can use `greater_equal_modulus()` here but not in `sigverify_mod_exp_otbn.c`.

**Review by alphan — APPROVED** [review]

File: sw/device/silicon_creator/lib/sigverify_mod_exp_ibex.c (hunk 2 above)
Comment (jadephilipoom) [reply sw/device/silicon_creator/lib/sigverify_mod_exp_ibex.c:171]: True! I'll change this one.

File: sw/device/lib/base/memory.c (hunk 1 above)
Comment (jadephilipoom) [reply sw/device/lib/base/memory.c:45]: Hmm, the rest of the file has `if !defined`  (in fact, this line is just moved, not added) -- should I change the whole file?

**Comment by jadephilipoom** [comment]
Will probably conflict with #10141; I'll rebase the remaining one after one is merged. (Even if git doesn't think it conflicts, I think it would be a good idea to rebase and run the CI again.)

**Pull request merged**